plugins/preview/plugin.js

```diff
@@ -74,27 +78,45 @@
 				window._cke_htmlToLoad = eventData.dataValue;
 			}
 
-			previewWindow = window.open( previewUrl, null, generateWindowOptions( windowDimensions ) );
+			nativePreviewWindow = window.open( previewUrl, null, generateWindowOptions( windowDimensions ) );
+			previewWindow = new CKEDITOR.dom.window( nativePreviewWindow );
 
 			// For IE we want to assign whole js stored in previewLocation, but in case of
 			// popup blocker activation oWindow variable will be null (https://dev.ckeditor.com/ticket/11597).
-			if ( previewLocation && previewWindow ) {
-				previewWindow.location = previewLocation;
+			if ( previewLocation && nativePreviewWindow ) {
+				nativePreviewWindow.location = previewLocation;
 			}
 
 			if ( !window._cke_htmlToLoad ) {
-				doc = previewWindow.document;
+				doc = nativePreviewWindow.document;
 
 				doc.open();
 				doc.write( eventData.dataValue );
 				doc.close();
 			}
 
-			return new CKEDITOR.dom.window( previewWindow );
+			if ( callback ) {
+				nativePreviewWindow.fireCallback = function() {
+					// In several browsers (e.g. Safari or Chrome on Linux) print command
+					// seems to be blocking loading of the preview page. Because of that
+					// print must be performed after the document is complete.
+					if ( !CKEDITOR.env.ie && nativePreviewWindow.document.readyState === 'complete' ) {
+						callback( previewWindow );
+					}
+
+					if ( CKEDITOR.env.ie ) {
+						callback( previewWindow );
+					}
+				};
+
+				nativePreviewWindow.fireCallback();
```

Comment:
Invoking it here means that in IE callback is invoked immediately, without waiting for `load` event (due to the second `if`).

Probably we will need to duplicate check of `readyState` outside of `fireCallback`.